Resources/Prototypes/Entities/Structures/Machines/deep_fryer.yml

```diff
@@ -0,0 +1,169 @@
+- type: entity
+  id: KitchenDeepFryer
+  parent: [ BaseMachinePowered, ConstructibleMachine ]
+  name: deep fryer
+  description: An industrial deep fryer. A big hit at state fairs!
+  components:
+  - type: Transform
+    noRot: false
+  - type: Sprite
+    netsync: false
+    sprite: Structures/Machines/deep_fryer.rsi
+    layers:
+      - state: off-0
+      - map: ["enum.SolutionContainerLayers.Fill"]
+        state: off-1
+  - type: AmbientSound
+    volume: -4
+    range: 4
+    enabled: false
+    sound:
+      path: /Audio/Ambience/Objects/deepfryer_sizzling.ogg
+  - type: SolutionContainerVisuals
+    maxFillLevels: 8
+    fillBaseName: off-
+  - type: Anchorable
+  - type: Pullable
+  - type: Rotatable
+    rotateWhilePulling: false
+  - type: Physics
+    bodyType: Static
+  - type: Climbable
+  - type: Fixtures
+    fixtures:
+    - shape:
+        !type:PhysShapeAabb
+        bounds: "-0.30,-0.25,0.30,0.45"
+      density: 600
+      mask:
+      - MachineMask
+      layer:
+      - MachineLayer
+  - type: DeepFryer
+    blacklist:
+      components:
+      # The classic.
+      - NukeDisk
+      # SliceableFood is handled easily enough, but there's not much to be
+      # gained by doing special handling for Stacks, especially since most
+      # of the items that use Stack aren't even remotely edible.
+      - Stack
+    whitelist:
+      components:
+      # It's what meat is.
+      - BodyPart
+      # Some clothes, shoes, uniforms.
```

Comment:
just have all clothing, should be fine

Reply:
That would mean the chief engineer's boots would be able to be turned into food.

Reply:
uhhh ok we'll figure stuff out later